main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+func main() {
+	lambda.Start(HandleLambdaEvent)
+}
+
+func HandleLambdaEvent() {
+	var r RealRunner
+	runReport(r)
+}
+
+func runReport(r Runner) {
+	log.SetFlags(0)
+
+	session := session.Must(session.NewSession())
+
+	if err := r.Setup(session); err != nil {
+		log.Printf("Setup error: %v", err)
+		return
+	}
+
+	dryRun := isDryRun()
+	if err := r.Run(dryRun); err != nil {
+		log.Printf("Run error: %v", err)
+		return
+	}
+
+	if !dryRun {
+		if err := r.Store(session); err != nil {
+			log.Printf("Store error: %v", err)
+			return
+		}
+	}
+}
+
+func isDryRun() bool {
+	dryRun, err := strconv.ParseBool(os.Getenv("GHTOOL_DRY_RUN"))
+	if err != nil {
+		return true
+	}
+	return dryRun
+}
+
+type Runner interface {
+	Setup(*session.Session) error
+	Run(bool) error
+	Store(*session.Session) error
+}
+type RealRunner struct{}
+
+func (r RealRunner) Setup(session *session.Session) error {
+	svc := ssm.New(session)
+
+	ssmPath := os.Getenv("GHTOOL_PARAM_NAME")
```

Comment:
I would probably use a different prefix than `GHTOOL_` as it is not consumed by the CLI tool. and it would be nice if the name also reflects what is inside the parameter.